src/yunohost/app.py

```diff
@@ -81,37 +82,69 @@ def app_listlists():
 
 def app_fetchlist(url=None, name=None):
     """
-    Fetch application list from app server
+    Fetch application list(s) from app server
 
     Keyword argument:
         name -- Name of the list (default yunohost)
         url -- URL of remote JSON list (default https://app.yunohost.org/official.json)
 
     """
     # Create app path if not exists
-    try: os.listdir(repo_path)
-    except OSError: os.makedirs(repo_path)
+    if not os.path.exists(repo_path):
+        os.makedirs(repo_path)
 
-    if url is None:
-        url = 'https://app.yunohost.org/official.json'
-        name = 'yunohost'
```

Comment:
As you removed default url/name, I wonder when this list is initially added. We may fix postinstall.